hydromt_delft3dfm/dflowfm.py

```diff
@@ -1487,6 +1458,385 @@ def setup_1dboundary(
             da_out, name=f"boundary1d_{da_out.name}_{branch_type}"
         )  # FIXME: this format cannot be read back due to lack of branch type info from model files
 
+    def _read_forcing_geodataset(
+        self, forcing_geodataset_fn: Union[str, Path], forcing_name: str = "discharge"
+    ):
+        """Read forcing geodataset."""
+
+        refdate, tstart, tstop = self.get_model_time()  # time slice
+
+        if (
+            forcing_geodataset_fn is not None
+            and self.data_catalog[forcing_geodataset_fn].data_type == "GeoDataset"
+        ):
+            da = self.data_catalog.get_geodataset(
+                forcing_geodataset_fn,
+                geom=self.region.buffer(
+                    1000
+                ),  # only select data within region of interest (large region for forcing)
+                variables=[forcing_name],
+                time_tuple=(tstart, tstop),
+            ).rename(forcing_name)
+            # error if time mismatch
+            if np.logical_and(
```

Comment:
Does this happen? Now that I think I think I had the issue with wflow. Might be an idea to have better support in core then ie an option to raise error when reading if time mismatch or no error but return a flag or dataset actual (start, end) tuple. What do you think?

Reply:
Csv or netcdf maybe still an issue for core? That if a time_tuple is passed to any of the get_data we should do an extra check in core to make sure the slicing happened?

Reply:
Yes, true! I added this request to hydromt issues.: https://github.com/Deltares/hydromt/issues/665